qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -412,16 +424,93 @@ def _default_options(cls):
                 curve.ParameterRepr("f_err", "stark_ferr", "Hz"),
             ],
             data_subfit_map={
-                "Xpos": {"series": "X", "direction": "pos"},
-                "Ypos": {"series": "Y", "direction": "pos"},
-                "Xneg": {"series": "X", "direction": "neg"},
-                "Yneg": {"series": "Y", "direction": "neg"},
+                "Fpos": {"direction": "pos"},
+                "Fneg": {"direction": "neg"},
             },
             plotter=ramsey_plotter,
         )
 
         return options
 
+    def _format_data(
+        self,
+        curve_data: curve.ScatterTable,
+    ) -> curve.ScatterTable:
+        columns = list(curve_data.columns)
+
+        i = 0
+        # Reassign model class.
+        # These data are actually P1 value of Ramsey X, Y quadrature,
+        # but the data is automatically classified based on the fit models,
+        # which are for the synthetic phase data in this analysis.
+        # This yields false classification.
+        for series in ("X", "Y"):
+            for direction in ("pos", "neg"):
+                rows = (curve_data.series == series) & (curve_data.direction == direction)
+                curve_data.loc[rows, "model_name"] = f"{series}{direction}"
+                # Assign random model id which will be invalidated later.
```

Comment:
This works but it feels like it is breaking the rules of the system. I wonder if the data processing steps could be adjusted to account for this. Looking at the steps I listed out in [an earlier comment](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1243#discussion_r1376794920), I think the model classification could occur later. Since `_run_data_processing()` saves the metadata for each measurement, it could group data by these metadata to do the `_format_data` averaging without assigning to a model ID (`model_id` could be `NA` for everything at this point). Then there could be a `_classify_models` hook that matches the models to the formatted data. In the case of `StarkRamseyXYAmpScanAnalysis`, it could call `super()._format_data()` and the add in the phase data. It could then either override `_classify_models()` or it could use different metadata for the phase data than the XY so that the default `_classify_models()` only mapped the phase data to models.

Reply:
Thanks. Delaying model mapping makes sense. Done in [bdd83d1](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1243/commits/bdd83d1c65731f4ef3ebe485ba4282e71f10f136) with a bit different approach.